### ISSUE TITLE
Multiple projects type per server

### ISSUE THREAD

**Issue by jeffmaury** [LucasBullen/Project-Provisioning-Protocol#6]
I can't see where the server is advertising the types of projects supported. For a single language, we may want to provide several different types of projects in a single server. Am I missing something

**Comment by mickaelistria** [comment]
I don't understand the request. Can you please elaborate on what you mean by "project types"? Examples would be welcome.

**Comment by jeffmaury** [comment]
Let's say I want to provide webservices Java type. I could have one for JAXRS and another one for JAXWS

**Comment by mickaelistria** [comment]
what would be the provisioning server? The same one for both (like JBoss Forge)?
Or do you want multiple wizards in UI backing to the same provisioning service but with some pre-sets?

**Comment by jeffmaury** [comment]
Yes I want a single server for both project types but from the UI point of view two entry points:

- create JAXRS project
- create JAXWS project

**Comment by mickaelistria** [comment]
Ok, that's noted as an interesting topic to evaluate soon and propose some solutions for.

More questions:
* Would it be simpler to have this ability on client side, or assume that this could be the same server, but started with different flag to restrict the scope? If the later, it's profitable to all clients. So it'd be `my-java-webservice-provisioner` to choose between both, `my-java-webserver-provisioner -jaxrs` to restrict to jaxrs and so on.
* If those are pre-selected at some point, do they replace templates (ie do we still need templates in the wizard).

My first impression is that it seems better to handle it on server side, and to be able to feed the provisioner with some options that would be sent on `initialize` to the server to restrict its goal and offer.

**Comment by LucasBullen** [comment]
Thoughts on the solution in the [multiProjectTypeServer branch](https://github.com/LucasBullen/Project-Provisioning-Protocol/commit/845bdfdbe45784ebb44f0f7912b632a48ee5bbf1)?

**Comment by mickaelistria** [comment]
I don't think it's a satisfactory solution. The protocol needs to focus on what the *user* needs to provision projects, keeping a vocabulary that's user oriented and specific to provisioning domain (avoiding widgets and so on). The proposal in the branch introduce some requests in the protocol for the sake of integration, but this is not something that's really targeting usage.
We need to stick to what's minimal to be fully usable, and the comments I've done above about just using a flag when connecting/starting the provisioning server seems to cover the use-case decently.
I suggest we delay this kind of changes until we have a really well qualified use-case that we didn't find a simpler way to cover.
